modules/peridynamics/src/userobjects/SingularShapeTensorEliminatorUserObjectPD.C

```diff
@@ -119,41 +115,17 @@ SingularShapeTensorEliminatorUserObjectPD::checkShapeTensorSingularity(const Ele
     if (_dim == 2)
       shape_tensor(2, 2) = 1.0;
 
-    if (_formulation == "CONVENTIONAL")
-    {
-      for (unsigned int nb = 0; nb < neighbors.size(); ++nb)
-        if (_bond_status_var->getElementalValue(_pdmesh.elemPtr(bonds[nb])) > 0.5)
-        {
-          vol_nb = _pdmesh.getPDNodeVolume(neighbors[nb]);
-          origin_vec_nb = *_pdmesh.nodePtr(neighbors[nb]) - *elem->node_ptr(nd);
-          weight_nb = horizon_nd / origin_vec_nb.norm();
-
-          for (unsigned int k = 0; k < _dim; ++k)
-            for (unsigned int l = 0; l < _dim; ++l)
-              shape_tensor(k, l) += weight_nb * origin_vec_nb(k) * origin_vec_nb(l) * vol_nb;
-        }
-    }
-    else if (_formulation == "BOND_HORIZON")
-    {
-      unsigned int nb_index =
-          std::find(neighbors.begin(), neighbors.end(), elem->node_id(1 - nd)) - neighbors.begin();
-      std::vector<dof_id_type> dg_neighbors =
-          _pdmesh.getDefGradNeighbors(elem->node_id(nd), nb_index);
-
-      for (unsigned int nb = 0; nb < dg_neighbors.size(); ++nb)
-        if (_bond_status_var->getElementalValue(_pdmesh.elemPtr(bonds[dg_neighbors[nb]])) > 0.5)
-        {
-          vol_nb = _pdmesh.getPDNodeVolume(neighbors[dg_neighbors[nb]]);
-          origin_vec_nb = *_pdmesh.nodePtr(neighbors[dg_neighbors[nb]]) - *elem->node_ptr(nd);
-          weight_nb = horizon_nd / origin_vec_nb.norm();
+    for (unsigned int nb = 0; nb < neighbors.size(); ++nb)
```

Comment:
Is there no need for the different 'formulation' options any more?

Reply:
I updated in _ComputeStrainBaseNOSPD_ that when the bond-associated shape tensor is singular we restore it to the conventional shape tensor and use the conventional deformation gradient for the rest simulation as long as it's nonsingular. This is more reasonable since we need to break all the bonds associated with a singular shape tensor.